cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/ProgramRuntimeProviderLoader.java

```diff
@@ -87,7 +68,7 @@ public ProgramRuntimeProviderLoader(CConfiguration cConf) {
   @Nullable
   public ProgramRuntimeProvider get(ProgramType programType) {
     try {
-      ProgramRuntimeProvider provider = programRunnerProviderCache.get(programType);
+      ProgramRuntimeProvider provider = programRuntimeExtensionLoader.getExtension(programType);
       if (provider != NOT_SUPPORTED_PROVIDER) {
```

Comment:
The `NOT_SUPPORTED_PROVIDER` is a private field defined in this class, how is it possible for `programRuntimeExtensionLoader.getExtension` returns the same instance??